docs/design/configs.md

```diff
@@ -0,0 +1,130 @@
+# Generic Config Object
```

Comment:
I definitely do not want to call this "Config". That term is too overloaded. Also, we don't abbreviate in the API.

I suggest "Parameters", to convey the intended usage -- this isn't intended to be a fully generic key-value store, but to provide configuration parameters to be consumed by applications, including system components.